services/QuillLMS/app/models/activity_session.rb

```diff
@@ -80,16 +95,17 @@ class LongTimeTrackingError < StandardError; end
   belongs_to :user
 
   before_create :set_state
-  before_save   :set_completed_at, :set_activity_id
+  before_save :set_completed_at, :set_activity_id
 
-  after_save    :determine_if_final_score, :update_milestones, :increment_counts
+  after_save :determine_if_final_score, :update_milestones, :increment_counts
   after_save :record_teacher_activity_feed, if: [:saved_change_to_completed_at?, :completed?]
+  after_save :save_user_pack_sequence_items, if: -> { saved_change_to_state? || saved_change_to_visible? }
 
   after_commit :invalidate_activity_session_count_if_completed
 
-  after_destroy :save_user_pack_sequence_items
+  after_save :trigger_events
```

Comment:
This becomes an `after_save` instead of an `around_save` because we don't do anything before the `yield` call, right?

Reply:
Correct.  There used to be a an assignment [before](https://github.com/empirical-org/Empirical-Core/commit/c596155ef2f998910f773ebdf82d6cc461d5a0df#diff-abe742d8f2a785c80119134a821bb01ab163ccd9041d1f329aefc9fe0401b075L262) yield, but now that is no longer there so it should just be an `after_save`.